### PR TITLE
Make testhost.x86 large address aware

### LINKED ISSUE #1985: testhost.x86 is not large address aware
Currently testhost.x86 is compiled with platform set to x86:

https://github.com/Microsoft/vstest/blob/53c0a341b07c1f224a31b2378e696f535a1f6080/src/testhost.x86/testhost.x86.csproj#L12

This compilation setting causes the application to omit the large address aware setting, which causes problems on larger multi-core machines (I'm observing frequent OOM errors on 2990WX).

The correct setting for an x86 build which is large address aware is a combination of the AnyCPU platform and the Prefer32Bit setting. See Microsoft/perfview@1582d04aefd7869a1ea6e1530c49ed2e6a1c5939 for an example of this configuration.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #1985

### CODE REVIEW

File: src/testhost.x86/testhost.x86.csproj [hunk 1]
```diff
@@ -9,8 +9,9 @@
     <TargetFrameworks>netstandard1.4;netcoreapp1.0;net451</TargetFrameworks>
     <TargetFrameworks Condition=" '$(DotNetBuildFromSource)' == 'true' ">netstandard1.4;netcoreapp1.0</TargetFrameworks>
     <WarningsAsErrors>true</WarningsAsErrors>
-    <PlatformTarget>x86</PlatformTarget>
-    <OutputType Condition="'$(TargetFramework)' != 'netstandard1.4'">Exe</OutputType>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <Prefer32Bit>true</Prefer32Bit>
+    <OutputType>Exe</OutputType>
```
Comment (mayankbansal018) [inline src/testhost.x86/testhost.x86.csproj]: please add condition TargetFramework != NS1.4, as we NS1.4 binary for running UWP tests, & we can't have it as exe.

Comment (sharwell) [reply src/testhost.x86/testhost.x86.csproj]: ➡️ netstandard1.4 and netcoreapp1.0 both produce a .dll as output under the current settings. The output type cannot be conditionally set without breaking LAA.

**Comment by mayankbansal018** [comment]
@sharwell , I'm trying this out locally, making sure it doesn't break UWP scenario, will get back to you by eod.

**Comment by sharwell** [comment]
@mayankbansal018 Thank you. I was expecting to need _some sort_ of additional change since this pull request likely changes the output directory for these assemblies, but I was unable to decipher the build output to determine if/what the specifics were.

**Comment by mayankbansal018** [comment]
@sharwell the PR doesn't change the output drop location, & I validated that even if we place testhost.dll compiled as exe for UWP projects, the tests still run fine.

Just a query regarding your comment https://github.com/Microsoft/vstest/issues/1985#issuecomment-483839381, do you mean the current testhost with LAA enabled after your changes crashes, or the one w/o LAA

**Comment by sharwell** [comment]
> the PR doesn't change the output drop location

Nice 👍 

> Just a query regarding your comment #1985 (comment), do you mean the current testhost with LAA enabled after your changes crashes, or the one w/o LAA

If I enable Server GC in the testhost.x86 that ships with Visual Studio (is not LAA), it runs out of memory.

> **This branch is out-of-date with the base branch**
> Merge the latest changes from `master` into this branch. 

This is a GitHub repository configuration error that I can help with. I'll send you the details on MS Teams.

**Comment by mayankbansal018** [comment]
@sharwell , can you please merge the master into this branch
Generally git gives an option to merge master, but don't know why it is not coming on this PR

**Comment by sharwell** [comment]
> @sharwell , can you please merge the master into this branch
> Generally git gives an option to merge master, but don't know why it is not coming on this PR

I sent you a message on MS Teams internally regarding this configuration error with the details on how to resolve it.

**Review by mayankbansal018 — APPROVED** [review]

**Pull request merged**